packages/teleterm/src/ui/services/workspacesService/workspacesService.ts

```diff
@@ -75,22 +76,30 @@ export class WorkspacesService extends ImmutableStore<WorkspacesState> {
   getWorkspaceDocumentService(
     clusterUri: string
   ): DocumentsService | undefined {
-    if (!this.documentsServicesCache.has(clusterUri)) {
+    // If the passed clusterUri is a leaf cluster URI, create a root cluster URI from it.
+    // Otherwise just use clusterUri.
+    const { leafClusterId, rootClusterId } =
```

Comment:
I'm afraid that it creates inconsistency. In other methods `clusterUri` parameter means `rootClusterUri` and only this method would be different.
What would you say for using `routing.parseClusterUri` in `commandLauncher`? I can see it is used there anyway.

Reply:
Alright, I'll change it then and close my other PR. Let's do the parsing in `commandLauncher` and see if we're going to run into more places which would like to use leaf cluster URI as well.

Reply:
I like it even better, because I renamed the `clusterUri` arg that the command launcher gets to `localClusterUri`, which signifies that the URI can be a root or a leaf cluster URI. 👍